subworkflows/local/contaminant_filter/main.nf

```diff
@@ -0,0 +1,324 @@
+        ch_versions = ch_versions.mix(BOWTIE2_ALIGN_TRNA.out.versions)
+
+        // Obtain how many hits were contaminants
+        ch_bowtie = BOWTIE2_ALIGN_TRNA.out.log
+
+        STATS_GAWK_TRNA(ch_bowtie, [])
+        ch_versions = ch_versions.mix(STATS_GAWK_TRNA.out.versions)
+
+        // Remove meta.contaminant and collect all contaminant stats in a single channel
+        ch_filter_stats = ch_filter_stats
+                .mix(STATS_GAWK_TRNA.out.output
+                        .map{meta, stats -> return [[id:meta.id, single_end:meta.single_end], stats]}
+                        .ifEmpty(null))
+
+        // Assign clean reads to new channel
+        trna_reads = BOWTIE2_ALIGN_TRNA.out.fastq
+    }
+
+    trna_reads.set { cdna_reads }
+
+    // Define how to filter significant BLAT hits
+    ch_program = Channel.value('BEGIN{FS="\t"}{if(\$11 < 1e-5) print \$2;}').collectFile(name:"program.txt")
+
+    if (params.cdna) {
+        // Search which hairpin miRNAs are present in the cDNA data
+        BLAT_CDNA(ch_reference_hairpin, ch_cdna)
+        ch_versions = ch_versions.mix(BLAT_CDNA.out.versions)
+
+        // Extract the significant hits
+        GAWK_CDNA(BLAT_CDNA.out.psl, ch_program)
+        ch_versions = ch_versions.mix(GAWK_CDNA.out.versions)
+
+        // Get only unique elements of the list
+        ch_pattern = GAWK_CDNA.out.output
+                .map { meta, file -> file.text.readLines() }
+                .flatten()
+                .unique()
+                .collectFile(name: 'ch_hairpin_cDNA_unique.txt', newLine: true)
+
+        // Remove the hairpin miRNAs from the cDNA data
+        SEQKIT_GREP_CDNA(ch_cdna, ch_pattern)
+        ch_versions = ch_versions.mix(SEQKIT_GREP_CDNA.out.versions)
+
+        // Previous original code:
+        INDEX_CDNA ( SEQKIT_GREP_CDNA.out.filter )
+        ch_versions = ch_versions.mix(INDEX_CDNA.out.versions)
+
+        // Add meta.contaminant to input reads channel
+        trna_reads = trna_reads.map{meta, fastq -> return [[id:meta.id, contaminant: "cDNA", single_end:meta.single_end], fastq]}
+
+        // Map which reads are cDNA
+        BOWTIE2_ALIGN_CDNA(trna_reads, INDEX_CDNA.out.index.first(), [[],[]], true, false)
+        ch_versions = ch_versions.mix(BOWTIE2_ALIGN_CDNA.out.versions)
+
+        // Obtain how many hits were contaminants
+        STATS_GAWK_CDNA(BOWTIE2_ALIGN_CDNA.out.log, [])
+        ch_versions = ch_versions.mix(STATS_GAWK_CDNA.out.versions)
+
+        // Remove meta.contaminant and collect all contaminant stats in a single channel
+        ch_filter_stats = ch_filter_stats
+                .mix(STATS_GAWK_CDNA.out.output
+                        .map{meta, stats -> return [[id:meta.id, single_end:meta.single_end], stats]}
+                        .ifEmpty(null))
+
+        // Assign clean reads to new channel
+        cdna_reads = BOWTIE2_ALIGN_CDNA.out.fastq
+    }
+
+    cdna_reads.set { ncrna_reads }
```

Comment:
```suggestion
    ncrna_reads = cdna_reads
```

Mayke sure to be consistent here for readabilityu, you're doing `ch_versions` everywhere so I assume this is preferred, and on line 168 above too